In [1]:
import pandas as pd

In [2]:


# Загрузим файл данных
file_path = '/content/sample_data/data.csv'
data = pd.read_csv(file_path)

# Просмотрим первые несколько строк данных
data.head()

Test Age_Group    Status
0  Positive     Young  Infected
1  Positive     Young  Infected
2  Positive     Young  Infected
3  Positive       Old  Infected
4  Positive       Old  Infected

In [3]:
# Преобразуем категориальные переменные в числовые значения
from sklearn.preprocessing import LabelEncoder

# Инициализируем кодировщики для категориальных признаков
label_encoder = LabelEncoder()

# Преобразуем 'Test', 'Age_Group', и 'Status' в числовые значения
data['Test'] = label_encoder.fit_transform(data['Test'])  # Positive: 1, Negative: 0
data['Age_Group'] = label_encoder.fit_transform(data['Age_Group'])  # Young: 1, Old: 0
data['Status'] = label_encoder.fit_transform(data['Status'])  # Infected: 1, Not Infected: 0

# Проверим преобразованные данные
data.head()

Test  Age_Group  Status
0     1          1       0
1     1          1       0
2     1          1       0
3     1          0       0
4     1          0       0

In [4]:
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score

# Разделим данные на признаки и целевую переменную
X = data[['Test', 'Age_Group']]  # Признаки
y = data['Status']  # Целевая переменная

# Разделим данные на обучающую и тестовую выборки
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Инициализируем модель Наивного Байеса
nb_model = GaussianNB()

# Обучаем модель
nb_model.fit(X_train, y_train)

# Прогнозируем на тестовой выборке
y_pred = nb_model.predict(X_test)

# Оценим точность модели
accuracy = accuracy_score(y_test, y_pred)
accuracy

0.6666666666666666

*  Модель Наивного Байеса для классификации показала точность 66.67% на тестовой выборке.

*  Это значение можно улучшить, исследуя другие методы подготовки данных, например, дополнительные признаки или другие подходы к обработке категориальных данных. Однако для текущей задачи, наивная модель дает базовое представление о качестве классификации